llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -747,6 +747,65 @@ void VPRegionBlock::execute(VPTransformState *State) {
   State->Instance.reset();
 }
 
+InstructionCost VPBasicBlock::cost(ElementCount VF, VPCostContext &Ctx) {
+  InstructionCost Cost = 0;
+  for (VPRecipeBase &R : Recipes)
+    Cost += R.cost(VF, Ctx);
+  return Cost;
+}
+
+InstructionCost VPRegionBlock::cost(ElementCount VF, VPCostContext &Ctx) {
+  if (!isReplicator()) {
+    InstructionCost Cost = 0;
+    Cost += Ctx.getLoopExitCost(VF);
```

Comment:
Could all cost pre-computations take place in one place, either here in loop-region.cost or in LAP.cost above?